src/univariate/discrete/negativebinomial2.jl

```diff
@@ -0,0 +1,121 @@
+"""
+Alternative location-scale parameterization of the negative binomial distribution
+in terms of location `μ` and overdispersion `ϕ`.
+
+```math
+P(X = n) = {n + \\phi - 1 \\choose n} (\\frac{\\mu}{\\mu + \\phi})^n (\\frac{\\phi}{\\mu + \\phi})^\\phi, \\quad \\text{for } n = 0,1,2,\\ldots
+```
+
+In terms of the mixture definition, this corresponds to:
+```math
+n \\sim \\text{Poisson}(z), \\quad
+z \\sim \\text{Gamma}(\\phi, \\frac{\\mu}{\\phi})
+```
+This provides a clear interpretation of the negative binomial as an overdispersed
+Poisson with expected location 𝔼[z] = μ and var[z] = μ + μ²/ϕ. It also has advantages in that
+its pmf can be expressed in terms of μ = exp(η), such that it can natively be
+parameterized on the log scale.
+
+```julia
+NegativeBinomial2()        # distribution with μ = 1.0 and ϕ = 1.0
+NegativeBinomial2(μ, ϕ)    # distribution with location μ and overdispersion ϕ
+
+convert(NegativeBinomial{T}, d)         # Parametric conversion to NegativeBinomial
+convert(NegativeBinomial2Log{T}, d)     # Parametric conversion to NegativeBinomial2Log
+convert(NegativeBinomial3{T}, d)        # Parametric conversion to NegativeBinomial3
+```
+
+External links:
+* [Negative binomial distribution (alternative paramerization), Stan](https://mc-stan.org/docs/2_29/functions-reference/nbalt.html)
+Note: the definition above uses the scale parameterization of the Gamma distribution
+(matching Distributions.jl), whereas Stan uses the inverse scale parameterization.
+
+See also: [`NegativeBinomial2Log`](@ref), [`NegativeBinomial3`](@ref)
+"""
+struct NegativeBinomial2{T<:Real} <: DiscreteUnivariateDistribution
+    μ::T
+    ϕ::T
+    NegativeBinomial2{T}(μ::T, ϕ::T) where {T<:Real} = new{T}(μ, ϕ)
+end
+
+function NegativeBinomial2(μ::T, ϕ::T; check_args::Bool=true) where {T<:Real}
+    @check_args NegativeBinomial2 (μ, μ > zero(μ)) (ϕ, ϕ > zero(ϕ))
+    return NegativeBinomial2{T}(μ, ϕ)
+end
+
+NegativeBinomial2(μ::Real, ϕ::Real; check_args::Bool=true) = NegativeBinomial2(promote(μ, ϕ)...; check_args=check_args)
+NegativeBinomial2(μ::Real; check_args::Bool=true) = NegativeBinomial2(μ, one(μ); check_args=check_args)
+NegativeBinomial2() = NegativeBinomial2{Float64}(1.0, 1.0)
```

Comment:
For distributions with a "standard" form, e.g. `Normal() == Normal(0, 1)` makes sense, but here I'm not sure in what situation you'd want to use a NegativeBinomial with mean 1 and variance 2... from a user point of view it seems better to say "you need to provide these arguments" than to hard-code these?
```suggestion
```

Reply:
This is a good point, which, incidentally, applies to all 3 new definitions. I added the zero-arg methods in analogy to `NegativeBinomial`'s zero-arg method. I am not particularly attached to them, hence, we could remove them? Thoughts @devmotion ?

Reply:
Seems reasonable. We can always add constructors with default arguments at a later point if we think they are really needed and/or useful.